packages/signature-controller/src/SignatureController.ts

```diff
@@ -250,6 +251,27 @@ export class SignatureController extends BaseControllerV2<
     return this.#typedMessageManager.getUnapprovedMessagesCount();
   }
 
+  /**
+   * Retrieves an array of all messages from TypedMessages and PersonalMessages.
+   *
+   * @returns The array of all messages from their proxies.
+   */
+  get allMessages(): unknown[] {
+    let messagesObject: any = {};
+
+    const allMessages = [
+      ...this.#typedMessageManager.getAllMessages(),
+      ...this.#personalMessageManager.getAllMessages(),
+      ...this.#messageManager.getAllMessages(),
+    ];
+
+    for (const message of allMessages) {
+      messagesObject = { ...messagesObject, ...message };
```

Comment:
Is this intentional? It appears to be just overwriting all the properties of the previous single message with the next one?